docs/infrastructure/microk8s/install.md

```diff
@@ -0,0 +1,274 @@
+4. Traffic forwarding must be configured on the machine so that the local microk8s cluster pods can reach the internet. This can be done by running `iptables -P FORWARD ACCEPT` (more info on this subject can be [found here](https://github.com/ubuntu/microk8s/tree/1.12#my-pods-cant-reach-the-internet-but-my-microk8s-host-machine-can)).
+5. Choose the microk8s release channel you want to install from.
+   1. The complete list of channels can be [found here](https://microk8s.io/docs/setting-snap-channel).
+   2. Assuming we want to install from the `stable` release channel, we export the following environment variable: `export MICROK8S_CHANNEL=stable`.
+6. We are finally in a state where we can install microk8s. That can be done via the `snap` package manager with: `snap install microk8s --classic --channel ${MICROK8S_CHANNEL}
+export PATH=$PATH:/snap/bin
+echo "--allow-privileged=true" >> /var/snap/microk8s/current/args/kube-apiserver
+microk8s.start && microk8s.status --wait-ready`
+
+### Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+   1. `sudo microk8s.enable dns && sudo microk8s.status --wait-ready --timeout 180`
+   2. `sudo microk8s.enable storage && sudo microk8s.status --wait-ready --timeout 180`
+   3. `sudo microk8s.enable ingress &&  sudo microk8s.status --wait-ready --timeout 180`
+   4. This will enable dns, storage and ingress on your microk8s cluster.
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster: `microk8s.stop && microk8s.start && microk8s.status --wait-ready
+sleep 10`. Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+   1. `microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s`
+   2. `microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner --timeout=300s`
+   3. `microk8s.kubectl wait -n default --for=condition=Ready pod -l app=default-http-backend --timeout=300s`
+   4. `microk8s.kubectl wait -n default --for=condition=Ready pod -l name=nginx-ingress-microk8s --timeout=300s`
+4. After runing these commands, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+### Installing Codacy
+
+#### Helm and Tiller
+
+1. First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+2. Assuming you would want to install `helm` version 2.16.3, export the following environment variable: `export HELM_VERSION=2.16.3`
```

Comment:
Should we state that we are supporting Helm 2 and not 3?

Reply:
yes!